src/Blorc.OpenIdConnect/Services/AccessTokenExpirationDelegatingHandler.cs

```diff
@@ -0,0 +1,35 @@
+﻿namespace Blorc.OpenIdConnect;
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+public class AccessTokenExpirationDelegatingHandler : DelegatingHandler
+{
+    private readonly IUserManager _userManager;
+
+    public AccessTokenExpirationDelegatingHandler(IUserManager userManager)
+    {
+        ArgumentNullException.ThrowIfNull(userManager);
+
+        _userManager = userManager;
+    }
+
+    protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+    {
+        ArgumentNullException.ThrowIfNull(request);
+
+        var user = await _userManager.GetUserAsync<User<Profile>>();
+        if (user?.ExpiresAt is not null)
+        {
+            var expiryDateTime = DateTimeOffset.FromUnixTimeSeconds(user.ExpiresAt).UtcDateTime;
+            if (DateTime.UtcNow > expiryDateTime)
```

Comment:
Please use ITimeProvider:

https://learn.microsoft.com/en-us/dotnet/api/system.timeprovider?view=net-9.0

If it's not yet available in older fx, we need to consider dropping these since the only supported fx are net 8 and 9.